coretex/cli/modules/utils.py

```diff
@@ -17,20 +17,45 @@
 
 from typing import List, Any, Optional, Callable
 from functools import wraps
+from importlib.metadata import version as getLibraryVersion
 
 import sys
 
 from py3nvml import py3nvml
 
 import click
+import requests
 
+from . import ui
 from ...utils.process import command
 
 
 def updateLib() -> None:
     command([sys.executable, "-m", "pip", "install", "--no-cache-dir", "--upgrade", "coretex"], ignoreStdout = True, ignoreStderr = True)
 
 
+def fetchLatestVersion() -> Optional[str]:
+    url = "https://pypi.org/pypi/coretex/json"
+    response = requests.get(url)
+    if response.ok:
+        data = response.json()
+        return str(data["info"]["version"])
+    else:
+        return None
+
+
+def checkLibVersion() -> None:
+    current = getLibraryVersion("coretex")
+    latest = fetchLatestVersion()
+
+    if latest is None:
+        return
+
+    if int(current[-3:]) < int(latest[-3:]):
```

Comment:
Major, minor, and patch versions should all be compared, not just the patch version. Patch version is not guaranteed to be the last 3 digits, what happens if this cannot be converted to integer?